docs/devel/kubectl-conventions.md

```diff
@@ -135,6 +133,151 @@ Updated: 8/27/2015
 * Use "TYPE" for the particular flavor of resource type accepted by kubectl, rather than "RESOURCE" or "KIND"
 * Use "NAME" for resource names
 
+## Command implementation conventions
+
+For every command there should be a `NewCmd<CommandName>` function that creates the command and returns a pointer to a `cobra.Command`, which can later be added to other parent commands to compose the structure tree. There should also be a `<CommandName>Options` struct with a variable to every flag and argument declared by the command (and any other variable required for the command to run). This makes tests and mocking easier. The struct ideally exposes three functions:
```

Comment:
> I like this proposal. @deads2k ?

I'm fine with pretty much any name for it.  I just want to make sure I end up with a value object containing all the information required to perform the action without any direct dependency on commands or flagsets.